File: flux-basic-java/src/test/java/com/example/app/AppTest.java
package com.example.app;

import com.example.app.user.api.GetUsers;
import io.fluxzero.sdk.test.TestFixture;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.annotation.DirtiesContext;

import static org.springframework.test.annotation.DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD;

@SpringBootTest(classes = App.class)
@DirtiesContext(classMode = AFTER_EACH_TEST_METHOD)
class AppTest {

	@Autowired TestFixture testFixture;

	@Test
	void createUser() {
		testFixture.whenCommand("/user/create-user.json")
				.expectEvents("/user/create-user.json");
	}

	@Test
	void getUsers() {
		testFixture.givenCommands("/user/create-user.json")
				.whenQuery(new GetUsers())
				.expectResult(r -> r.size() == 1);
	}
}

File: flux-basic-kotlin/src/test/kotlin/com/example/app/AppTest.kt
package com.example.app

import com.example.app.user.api.GetUsers
import com.example.app.user.api.model.UserProfile
import io.fluxzero.sdk.test.TestFixture
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.annotation.DirtiesContext

import org.springframework.test.annotation.DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD

@SpringBootTest(classes = [App::class])
@DirtiesContext(classMode = AFTER_EACH_TEST_METHOD)
class AppTest {

    @Autowired
    lateinit var testFixture: TestFixture

    @Test
    fun createUser() {
        testFixture.whenCommand("/user/create-user.json")
            .expectEvents("/user/create-user.json")
    }

    @Test
    fun getUsers() {
        testFixture.givenCommands("/user/create-user.json")
            .whenQuery(GetUsers())
            .expectResult<List<UserProfile>> { r: List<UserProfile> -> r.size == 1 }
    }
}